imagery/i.cca/main.c

```diff
@@ -169,13 +169,14 @@ int main(int argc, char *argv[])
      */
 
     samptot = 0;
-    for (i = 1; i <= nclass; i++) {
-        nsamp[i] = sigs.sig[i - 1].npoints;
+    for (i = 0; i < nclass; i++) {
+        nsamp[i] = sigs.sig[i].npoints;
         samptot += nsamp[i];
-        for (j = 1; j <= bands; j++) {
-            mu[i][j] = sigs.sig[i - 1].mean[j - 1];
-            for (k = 1; k <= j; k++)
-                cov[i][j][k] = cov[i][k][j] = sigs.sig[i - 1].var[j - 1][k - 1];
+        for (j = 0; j < bands; j++) {
+            mu[i][j] = sigs.sig[i].mean[j];
+            for (k = 0; k < j; k++) {
```

Comment:
try
```suggestion
            for (k = 0; k <= j; k++) {
```
because j is not the upper bound of k to be excluded but to be included, also with zero-based indexing.

Reply:
The functions in [stats.c](https://github.com/OSGeo/grass/blob/main/imagery/i.cca/stats.c) assume zero-based indices, thus I would expect different results in the sense that results with your PR (zero based indices) are correct whereas previous results (one based indices) were not correct.

You would need some independent, trusted alternative for Canonical components analysis in order to decide which results are correct. Regarding the code, your PR seems like a bugfix, previous results are probably not correct.

Reply:
Just for clarification, `cov` has been indexed previously in `main.c` as `cov[i][j][k]` which was wrong for one-based i, j, k because zero-based i, j, k are used in `stats.c`. With your PR, `cov` is consistently indexed with zero-based i, j, k, that could explain the differences.